### DISCUSSION TITLE
Click event inside the `Link`

### DISCUSSION THREAD

**Discussion by SupertigerDev** [solidjs/solid-router#158] [Q&A]
If i have something like this:
```jsx
<Link href={somePath}>
   <div>Click me to View</div>
   <div onClick={onClose}>Click me to close</div>
</Link>
```
How can I prevent the navigation from the `Link` component from happening when i click on the `Click me to close` div?

https://stackblitz.com/edit/github-imhxsg?file=src%2FApp.tsx%3AL26

Edit: In the stackblitz above, adding `preventDefault` to `onClose` Works it seems. But in the example below, It doesnt. Strange.
https://stackblitz.com/edit/github-imhxsg?file=src%2FApp.tsx

Edit: oops, i forgot to fork the project, so both links are the same ;( but the issue is still there

**Comment by ryansolid** [comment]
Does preventDefault on the event work:
```js
const onClose = (event) => {
    event.preventDefault();
    alert('clicked');
 };
```

**Reply by SupertigerDev** [reply]
Already tried that and `stopPropagation`, but it makes no difference :(

**Reply by SupertigerDev** [reply]
Nevermind, its working somehow. I swear it was not working on my project. I will keep experimenting. Thanks!

**Reply by SupertigerDev** [reply]
I dont understand. On my project, doing the exact same thing does not work. I updated to latest solidjs and router package hmm

**Reply by SupertigerDev** [reply]
Okay, In this example,  `preventDefault` does not seem to work:
https://stackblitz.com/edit/github-imhxsg?file=src%2FApp.tsx

**Reply by SupertigerDev** [reply]
I created an issue for this (https://github.com/solidjs/solid/issues/1178) as this seems like a bug. When using `preventDefault` within `Suspense`, it does not work.

**Reply by ryansolid** [reply]
Yeah that feels like a bug. Very strange why this would differ but timing must be impacted somehow.
